### PR TITLE
Uniform code and doc style

### PR DESCRIPTION
**Pull request by thiemowmde**
* Add function some level documentation.
* Logically `@throw` comes before `@return`.
* Add spaces in `if (` and such.

### CODE REVIEW

File: src/Deserializers/EntityDeserializer.php [hunk 1]
```diff
@@ -59,6 +59,11 @@ public function deserialize( $serialization ) {
 	 */
 	protected abstract function getPartiallyDeserialized( array $serialization );
 
+	/**
+	 * @param array $serialization
+	 *
+	 * @return Entity
```
Comment (JeroenDeDauw) [inline src/Deserializers/EntityDeserializer.php]: EntityDocument please

**Comment by JeroenDeDauw** [comment]
+1. If my comment is addressed, +2. Willing to make this change myself.

File: src/Deserializers/EntityDeserializer.php (hunk 1 above)
Comment (thiemowmde) [inline src/Deserializers/EntityDeserializer.php]: I tried but it's not that easy. The whole class depends on Entity. Please merge this and try to phase out Entity in an other patch.

Comment (JeroenDeDauw) [inline src/Deserializers/EntityDeserializer.php]: What do you mean with it not being that easy? What problem occurs when using this type hint?

Comment (thiemowmde) [inline src/Deserializers/EntityDeserializer.php]: It's not true. A few lines below the code returns an `Entity`.

Comment (JeroenDeDauw) [inline src/Deserializers/EntityDeserializer.php]: Then it is true, since Entity is a subtype of EntityDocument.

Comment (thiemowmde) [inline src/Deserializers/EntityDeserializer.php]: Ok, both is true. But this class is not called `EntityDocumentDeserializer`.

Comment (JeroenDeDauw) [inline src/Deserializers/EntityDeserializer.php]: Sure. So will you change it? Or let me change it? If not, that's fine, though then I'm sticking with my +1.

Comment (thiemowmde) [inline src/Deserializers/EntityDeserializer.php]: Change what? The class name?

**Comment by Benestar** [comment]
@thiemowmde Are you going to rebase this or should it be closed?

**Comment by thiemowmde** [comment]
Why should a 1 line conflict make a 100+ lines patch obsolete?

**Pull request merged**